cmd/main.go

```diff
@@ -130,43 +133,58 @@ func Execute() {
 	for _, plugin := range allPlugins {
 		subCommand, err := plugin.DefineCommand(channels.Items, channels.Errors)
 		if err != nil {
-			log.Fatal().Msg(fmt.Sprintf("error while defining command for plugin %s: %s", plugin.GetName(), err.Error()))
+			return fmt.Errorf("error while defining command for plugin %s: %s", plugin.GetName(), err.Error())
 		}
 		subCommand.GroupID = group
-		subCommand.PreRun = preRun
-		subCommand.PostRun = postRun
+		subCommand.PreRunE = preRun
+		subCommand.PostRunE = postRun
 		rootCmd.AddCommand(subCommand)
 	}
 
 	if err := rootCmd.Execute(); err != nil {
-		log.Fatal().Msg(err.Error())
+		return err
 	}
+
+	if report.TotalSecretsFound > 0 && ShowError("errors") {
+		return fmt.Errorf("")
+	}
+
+	return nil
 }
 
-func validateFormat(stdout string, reportPath []string) {
+func validateFormat(stdout string, reportPath []string) error {
 	r := regexp.MustCompile(outputFormatRegexpPattern)
 	if !(r.MatchString(stdout)) {
-		log.Fatal().Msgf(`invalid output format: %s, available formats are: json, yaml and sarif`, stdout)
+		return fmt.Errorf(`invalid output format: %s, available formats are: json, yaml and sarif`, stdout)
 	}
 
 	for _, path := range reportPath {
 		fileExtension := filepath.Ext(path)
 		format := strings.TrimPrefix(fileExtension, ".")
 		if !(r.MatchString(format)) {
-			log.Fatal().Msgf(`invalid report extension: %s, available extensions are: json, yaml and sarif`, format)
+			return fmt.Errorf(`invalid report extension: %s, available extensions are: json, yaml and sarif`, format)
 		}
 	}
+
+	return nil
 }
 
-func preRun(cmd *cobra.Command, args []string) {
-	validateFormat(stdoutFormatVar, reportPathVar)
+func preRun(cmd *cobra.Command, args []string) error {
+	if err := validateFormat(stdoutFormatVar, reportPathVar); err != nil {
+		return err
+	}
+
 	secrets, err := secrets.Init(includeRuleVar, excludeRuleVar)
 	if err != nil {
-		log.Fatal().Msg(err.Error())
+		return err
 	}
 
 	if err := secrets.AddRegexRules(customRegexRuleVar); err != nil {
-		log.Fatal().Msg(err.Error())
+		return err
+	}
+
+	if err := InitShouldIgnoreArg(ignoreOnExitVar); err != nil {
+		return err
 	}
```

Comment:
I think you need to move it up as you can, since it defines the error behavior, it must be as first as we can, before other potential errors.

Reply:
done